spatialprofilingtoolbox/workflow/common/squidpy.py

```diff
@@ -0,0 +1,97 @@
+"""Low-level calculations of squidpy metrics."""
+
+from typing import Any
+
+from numpy.typing import NDArray
+from pandas import DataFrame
+from squidpy.gr import spatial_neighbors, nhood_enrichment, co_occurrence, ripley
+from anndata import AnnData
+
+from spatialprofilingtoolbox.db.exchange_data_formats.metrics import PhenotypeCriteria
+from spatialprofilingtoolbox.workflow.common.cell_df_indexer import get_mask
+
+
+def describe_squidpy_feature_derivation_method():
+    return '''
+    Calculates nhood_enrichment, co_occurrence, and ripley from squidpy.gr using clusters derived
+    from the phenotypes provided. Reference db.squidpy_metrics.convert_df_to_anndata for the
+    clustering method.
```

Comment:
This is not meant to be an uncontrolled description string, it is supposed to exactly tell what feature values were intended to be computed, given the specific definite list of specifiers for the feature. E.g.:

```txt
For a given cell phenotype (first specifier), the average number of cells of a second phenotype (second specifier) within a specified radius (third specifier).
```
In the example, there are exactly 3 specifiers, and the description tells what was going on in real life if you get a given number as the feature value for a given sample.
This description becomes a data value in the database, not a docstring somewhere to be read by a developer. So references to implementation `squidpy.gr` and `db.squidpy_metrics...` are also against what was intended.
It seems there are 3 or 4 different things to write here, depending on which specific feature class is computed (Ripley, co-occurence, etc.).
We should write these after we are sure about which thing is the final feature value actually recorded. As discussed in another thread, a 5000-entry matrix is not an acceptable "value" for a summary feature, so there is some more work to do here.